app/services/resync_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class ResyncService < ApplicationService
+  attr_reader :document
+
+  def initialize(document)
+    @document = document
+  end
+
+  def call
+    if document.live_edition.present?
+      # Present to Publishing API & Asset Manager draft stack
+      PreviewService.call(document.live_edition)
+
+      # Publish document to live stack
+      #  (We can't use PublishService directly as it promotes
+      #  `current_edition` to `live_edition`. We just want to
+      #  resync the two with Publishing API, not change their
+      #  states.)
+      GdsApi.publishing_api_v2.publish(
+        document.content_id,
+        nil, # Sending update_type is deprecated (now in payload)
+        locale: document.locale,
+      )
```

Comment:
A thing we'll have to ponder between the preview and publish call is that one of them needs to tell the Publishing API that we're sending a republish.

Reply:
Ah, I meant for this to be a republish. Good spot.

Reply:
Just a few notes on that front.

Publishing API has a [/v2/republish](https://github.com/alphagov/publishing-api/blob/master/doc/api.md#post-v2contentcontent_idrepublish) endpoint. This is used to transition unpublished content back to published so cannot be used in this scenario.

We need to set an update_type of "republish" to the Publishing API. This can either be done through PutContent (the preferred way) or through the Publish method (the deprecated way). So if we can set it at the preview point I'd recommend that.

The whole aspect of `update_type: republish` is a bit quirky so don't be too surprised if we hit something a bit thorny.

Reply:
Ok - have updated the GdsApi helper method we call.

Reply:
>  through the Publish method (the deprecated way)

Should we edit https://github.com/alphagov/publishing-api/blob/master/doc/api.md#post-v2contentcontent_idrepublish to make it clear this approach is no longer recommended?

Reply:
Also this doc needs updating I think: https://github.com/alphagov/publishing-api/blob/master/doc/api.md#post-v2contentcontent_idpublish

> update_type (conditionally required, deprecated)
> This is deprecated, clients should instead set the update type when putting the content.

...but if I publish without setting an update type (`curl http://publishing-api.dev.gov.uk/v2/content/cdaebb8f-02ae-48ef-be0a-b718b49f9a23/publish -H "Content-Type: application/json" --request POST --data '{}'`) I get an error:

`{"error":{"code":422,"message":"update_type is required","fields":{"update_type":["is invalid"]}}}`

Reply:
> > through the Publish method (the deprecated way)
> 
> Should we edit https://github.com/alphagov/publishing-api/blob/master/doc/api.md#post-v2contentcontent_idrepublish to make it clear this approach is no longer recommended?

I think we have crossed wires here as this republish method is allowed and fine - it's just not for our needs.